### PR TITLE
CLI: resource commands

### PR DESCRIPTION
**Pull request by bergman**
- list
- show
- create
- edit

### CODE REVIEW

File: styx-cli/src/main/java/com/spotify/styx/cli/Main.java [hunk 1]
```diff
@@ -194,6 +197,29 @@ private void run() {
           }
           break;
 
+        case RESOURCE:
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: Starting to feel the smell of switch inside a switch. Maybe it's better to have a method for subcommand processing.  Can be done via another PR.

Comment (bergman) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: I fully agree. I wanted to modularize the different CLI components but wanted to keep this PR lean. Let's do it next!

**Comment by honnix** [comment]
Minor comments, otherwise 👍

File: styx-cli/src/main/java/com/spotify/styx/cli/Main.java [hunk 2]
```diff
@@ -528,6 +624,35 @@ public Subparser parser(Subparsers subCommands) {
...
+    private final String alias;
+    private final String description;
+
+    ResourceCommand(String alias, String description) {
+      this.alias = alias;
+      this.description = description;
+    }
+
+    public Subparser parser(Subparsers subCommands) {
+      Subparser subparser = subCommands
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: `final`. For `BackfillCommand` as well.

File: styx-cli/src/main/java/com/spotify/styx/cli/Main.java [hunk 3]
```diff
@@ -659,23 +659,21 @@ public Subparser parser(Subparsers subCommands) {
...
+        attrs.put(arg.getDest(),
+                  ParameterUtil.parseDateHour(value.toString()));
+      } catch (DateTimeParseException dateHourException) {
         try {
-          instant = ParameterUtil.parseDate(value.toString());
-        } catch (Exception ignoredInner) {
+          attrs.put(arg.getDest(),
+                    ParameterUtil.parseDate(value.toString()));
+        } catch (Exception dateException) {
+          throw new ArgumentParserException(
```
Comment (honnix) [inline styx-cli/src/main/java/com/spotify/styx/cli/Main.java]: 👍

**Pull request merged**